### PR TITLE
Faster array-unique implementation

### PR DESCRIPTION
**Pull request by pranaygp**
Using the array-unique package to find unique file names seemed rather slow. I think it has to do with the [expensive `Array.prototype.splice` calls](https://github.com/jonschlinkert/array-unique/blob/master/index.js#L23).

Using an ES6 trick with `Array.from(new Set(array))` I'm able to decrease that time quite a bit.

Running a test on an npm project (zeit/front), which has 411 file names in the unique array, is a lot faster (I'm able to reproduce this consistently):

```
> [debug] testing array-unique on files: 2.578ms
> [debug] testing Array from Set implementation on files: 0.555ms
```

As an added benefit, one less dependency :)

### CODE REVIEW

**Comment by pranaygp** [comment]
Interestingly, I added the set implementation as a test to `array-unique`'s benchmark suite, and it had a different story to tell:

```
array-unique   (master ⚡☡=)  yarn run benchmark
yarn run v1.2.1
$ node benchmark/index.js
Benchmarking: (3 of 3)
 · long
 · med
 · short

# benchmark/fixtures/long.js (76 bytes)
  array-unique-immutable:
  array-unique x 773 ops/sec ±9.08% (77 runs sampled)
  filter x 14.14 ops/sec ±1.61% (35 runs sampled)
  for-in x 449 ops/sec ±0.83% (86 runs sampled)
  linear-search-01 x 4.17 ops/sec ±3.06% (14 runs sampled)
  set x 594 ops/sec ±1.25% (82 runs sampled)
  while-hash x 516 ops/sec ±1.58% (82 runs sampled)
  while x 18.49 ops/sec ±1.14% (46 runs sampled)

  fastest is array-unique

# benchmark/fixtures/med.js (74 bytes)
  array-unique-immutable:
  array-unique x 21,383 ops/sec ±1.60% (85 runs sampled)
  filter x 10,088 ops/sec ±1.19% (85 runs sampled)
  for-in x 9,370 ops/sec ±0.83% (89 runs sampled)
  linear-search-01 x 2,796 ops/sec ±0.86% (87 runs sampled)
  set x 12,247 ops/sec ±0.84% (86 runs sampled)
  while-hash x 12,187 ops/sec ±1.43% (84 runs sampled)
  while x 17,535 ops/sec ±0.81% (86 runs sampled)

  fastest is array-unique

# benchmark/fixtures/short.js (73 bytes)
  array-unique-immutable:
  array-unique x 1,159,738 ops/sec ±1.26% (90 runs sampled)
  filter x 1,336,787 ops/sec ±0.61% (87 runs sampled)
  for-in x 346,310 ops/sec ±1.12% (90 runs sampled)
  linear-search-01 x 5,389,841 ops/sec ±0.67% (92 runs sampled)
  set x 628,843 ops/sec ±1.17% (84 runs sampled)
  while-hash x 371,995 ops/sec ±1.85% (85 runs sampled)
  while x 5,064,443 ops/sec ±0.85% (85 runs sampled)

  fastest is linear-search-01
✨  Done in 117.17s.
```

Testing with only strings in the array has the following results:

```
# benchmark/fixtures/strings.js (339 bytes)
  array-unique-immutable:
  array-unique x 23,141 ops/sec ±2.80% (86 runs sampled)
  filter x 2,543 ops/sec ±0.88% (88 runs sampled)
  for-in x 24,197 ops/sec ±2.13% (86 runs sampled)
  linear-search-01 x 2,726 ops/sec ±0.61% (88 runs sampled)
  set x 16,022 ops/sec ±0.85% (87 runs sampled)
  while-hash x 18,622 ops/sec ±0.51% (88 runs sampled)
  while x 2,417 ops/sec ±0.65% (84 runs sampled)

  fastest is for-in
```

(for more information: https://github.com/pranaygp/array-unique/tree/benchmarking/benchmark)

I'm now thoroughly confused

**Comment by pranaygp** [comment]
TL;DR forIn >> array-unique and forIn > Set 

From the previous tests, I figured for-in might be a better way to go w.r.t string. I did more tests, specifically using real filenames as the input set. I also ran tests with no duplicates, every element having a duplicate (the same array concat'd twice) and shuffling the duplicated list (probably the worst scenario).

my forIn implementation won (seems like a pretty straightforward coding interview question haha). I think this actually amkes sense given that we're caching everything in an object, and since we're dealing with objects instead of arrays/numbers as keys, JS doesn't have to do any extra work in serializing keys. For strings, in general, forIn seems to be the best.

If you're curious: 

```
array-unique x 73,779 ops/sec ±2.49% (81 runs sampled)
set x 51,384 ops/sec ±2.94% (81 runs sampled)
for in x 99,179 ops/sec ±3.07% (81 runs sampled)
better for in x 121,670 ops/sec ±2.62% (81 runs sampled)

Fastest is better for in

array-unique duplicated x 65,785 ops/sec ±2.40% (78 runs sampled)
set duplicated x 55,063 ops/sec ±2.13% (86 runs sampled)
for in duplicated x 109,488 ops/sec ±0.97% (88 runs sampled)
better for in duplicated x 127,666 ops/sec ±1.96% (88 runs sampled)

Fastest is better for in duplicated

array-unique shuffled x 72,387 ops/sec ±1.45% (86 runs sampled)
set shuffled x 54,278 ops/sec ±2.59% (83 runs sampled)
for in shuffled x 98,850 ops/sec ±2.79% (83 runs sampled)
better for in shuffled x 126,678 ops/sec ±2.31% (86 runs sampled)

Fastest is better for in shuffled
```

A (random) test from `now-cli` on `zeit/front`:
```
...
> [debug] locating files /Users/pranaygp/github/front: 15.909ms
set implementation: 0.894ms
for in implementation: 0.614ms
> [debug] Getting files: 25.051ms
> [debug] Need to compute hashes for 441 files
...
```

Ready ~to Ship~ for review! :)

**Comment by leo** [comment]
Thanks a lot! 😍
